lib/node_modules/@stdlib/math/base/special/lnf/src/main.c

```diff
@@ -0,0 +1,197 @@
+*
+* ```text
+* Copyright (C) 1993 by Sun Microsystems, Inc. All rights reserved.
+*
+* Developed at SunPro, a Sun Microsystems, Inc. business.
+* Permission to use, copy, modify, and distribute this
+* software is freely granted, provided that this notice
+* is preserved.
+* ```
+*/
+
+#include "stdlib/math/base/special/lnf.h"
+#include "stdlib/number/float32/base/to_word.h"
+#include "stdlib/number/float32/base/from_word.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/constants/float32/ninf.h"
+#include "stdlib/constants/float32/exponent_bias.h"
+#include "stdlib/constants/float32/exponent_mask.h"
+#include "stdlib/constants/float32/abs_mask.h"
+#include "stdlib/constants/float32/significand_mask.h"
+#include <stdint.h>
+
+static const float LN2_HI = 6.9313812256e-01f; // 0x3f317180
+static const float LN2_LO = 9.0580006145e-06f; // 0x3717f7d1
+static const float TWO25 = 3.355443200e+07f;   // 0x4c000000
+static const float ONE_THIRD = 0.33333333333333333f;
+
+/* Begin auto-generated functions. The following functions are auto-generated. Do not edit directly. */
+
+// BEGIN: polyval_p
+
+/**
+* Evaluates a polynomial.
+*
+* ## Notes
+*
+* -   The implementation uses [Horner's rule][horners-method] for efficient computation.
+*
+* [horners-method]: https://en.wikipedia.org/wiki/Horner%27s_method
+*
+* @param x    value at which to evaluate the polynomial
+* @return     evaluated polynomial
+*/
+static float polyval_p( const float x ) {
+	return 0.40000972152f + (x * 0.24279078841f);
+}
+
+// END: polyval_p
+
+// BEGIN: polyval_q
+
+/**
+* Evaluates a polynomial.
+*
+* ## Notes
+*
+* -   The implementation uses [Horner's rule][horners-method] for efficient computation.
+*
+* [horners-method]: https://en.wikipedia.org/wiki/Horner%27s_method
+*
+* @param x    value at which to evaluate the polynomial
+* @return     evaluated polynomial
+*/
+static float polyval_q( const float x ) {
+	return 0.66666662693f + (x * 0.28498786688f);
+}
+
+// END: polyval_q
+
+/* End auto-generated functions. */
+
+/**
+* Evaluates the natural-logarithm of a single-precision floating-point number.
+*
+* @param x    input value
+* @return     natural logarithm
+*
+* @example
+* float out = stdlib_base_lnf( 2.0f );
+* // returns ~0.693f
+*/
+float stdlib_base_lnf( const float x ) {
+	uint32_t uix;
+	int32_t ix;
+	float hfsq;
+	int32_t k;
+	int32_t i;
+	int32_t j;
+	float dk;
+	float t1;
+	float t2;
+	float xc;
+	float f;
+	float R;
+	float s;
+	float z;
+	float w;
+
+	if ( x == 0.0f ) {
+		return STDLIB_CONSTANT_FLOAT32_NINF;
+	}
+	if ( stdlib_base_is_nan( x ) || x < 0.0f ) {
+		return 0.0f / 0.0f; // NaN
+	}
+	stdlib_base_float32_to_word( x, &uix );
+	ix = (int32_t)uix;
+	xc = x;
+	k = 0;
+	// x < 2**-126
+	if ( ix < 0x00800000 ) {
+		if ( ( ix & STDLIB_CONSTANT_FLOAT32_ABS_MASK ) == 0 ) {
+			// log(+-0)=-inf
+			return -TWO25 / 0.0f;
+		}
```

Comment:
L119-121 already have a check for an input value of zero, so I don't think this is needed.

Reply:
In C, it may be better to get rid of L119 and L122 blocks, as computation of negative infinity will trigger floating-point exceptions, which may be desirable in native code. In JS, it should be fine to remove these blocks, but I'd do the inverse in C.

Reply:
That would match the reference implementation, so yes, removing the other blocks for C seems like a good solution.

Reply:
Removed both of the blocks.